app/src/main/java/pl/netigen/drumloops/rock/core/base/TypeParamsUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package pl.netigen.drumloops.rock.core.base
+
+import kotlinx.coroutines.*
+
+abstract class TypeParamsUseCase<out Type, in Params> {
```

Comment:
It will be better to have one abstract class UseCase<out Type, in Params> and use Unit where there are no parameters or result type